lib/redis/hash_ring.rb

```diff
@@ -85,5 +86,16 @@ def self.binary_search(ary, value)
       upper = ary.size - 1 if upper < 0
       upper
     end
+
+    private
+
+    def hash_for(key)
+      Zlib.crc32(key)
+    end
+
+    def server_hash_for(key)
+      hash = Digest::MD5.hexdigest(key)
+      hash[0..7].hex
```

Comment:
Hum, we're extracting the first 32bits right?

I think we can probably do this more efficiently with: `Digest::MD5.digest(key).unpack1("L>")`

```
Warming up --------------------------------------
                 hex    58.873k i/100ms
              unpack    80.146k i/100ms
Calculating -------------------------------------
                 hex    572.792k (± 4.2%) i/s -      2.885M in   5.046005s
              unpack    753.817k (±10.0%) i/s -      3.767M in   5.061550s

Comparison:
              unpack:   753817.0 i/s
                 hex:   572791.5 i/s - 1.32x  (± 0.00) slower
```

```ruby
# frozen_string_literal: true

# >> Digest::MD5.digest("key").unpack1("L>")
# => 1013844874
# >> Digest::MD5.hexdigest("key")[0..7].hex
# => 1013844874

require 'benchmark/ips'
require "digest/md5"
Benchmark.ips do |x|
  x.report("hex") { Digest::MD5.hexdigest("key")[0..7].hex }
  x.report("unpack") { Digest::MD5.digest("key").unpack1("L>") }
  x.compare!
end
```

Not a huge difference, but I think I find it cleaner anyway.

Reply:
Nice 👍 Updated with this change.